src/common/snippets/src/lowered/pass/iter_handler.cpp

```diff
@@ -97,14 +97,16 @@ bool TransformInnerSplitLoop::run(LinearIR& linear_ir, LinearIR::constExprIt beg
     bool modified = false;
     for (auto it = begin; it != end; ++it) {
         const auto& expr = *it;
-        const auto inner_loop_end = ov::as_type_ptr<op::LoopEndStatic>(expr->get_node());
+        const auto inner_loop_end = ov::as_type_ptr<op::LoopEnd>(expr->get_node());
         if (!inner_loop_end)
             continue;
         // There is already ExpandedLoopInfo
         const auto inner_loop_info = loop_manager->get_loop_info<ExpandedLoopInfo>(inner_loop_end->get_id());
         const auto inner_dim_idx = inner_loop_info->get_dim_idx();
         if (inner_dim_idx != current_dim_idx)
             continue;
+        // TODO [141735] : At the moment Splitted loops are not supported in dynamic case
+        OPENVINO_ASSERT(!inner_loop_end->has_dynamic_params(), "inner loop must be static in TransformInnerSplitLoop");
```

Comment:
Shouldn't we avoid calling `SplitLoops` pass on loops that have some dynamic inner loops?
Or even if any of the loops in LIR is dynamic?

In other words, if it is possible that this exception will be thrown (and the inference will be terminated by exception), then we should completely turn off this transformation in dynamic pipeline.

Reply:
Sounds logically! Totally agree, added check in `can_be_split` in `SplitLoops` pass.
Thanks!